hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemE2E.java

```diff
@@ -209,9 +209,10 @@ public void testFlushWithFileNotFoundException() throws Exception {
 
     fs.delete(testFilePath, true);
     assertFalse(fs.exists(testFilePath));
+    AbfsConfiguration configuration = this.getConfiguration();
 
-    intercept(FileNotFoundException.class,
-            () -> stream.close());
+    //With the new code, it would not trigger a call to the backend
```

Comment:
Could you please point me to the check in close() call that prevents server call ?

Also have the tests been run on these different configurations:
1. With appenblob feature enabled
- With and Without Namespace enabled account : fs.azure.enable.appendwithflush true and false
2. With appenblob feature disabled
- With and Without Namespace enabled account : fs.azure.enable.appendwithflush true and false

Reply:
I updated the comment with more details. You can also look at the code to understand the behavior. AbfsOutputStream close(). Yes I have tested the different configurations (appendblob, blockblob, append+flush enabled, append+flush disabled, with and without namespace disabled account.